drivers/vmwarevsphere/flags.go

```diff
@@ -278,3 +295,21 @@ func (f *FileAttrFlag) SetPerms(owner, group, perms int) {
 func (f *FileAttrFlag) Attr() types.BaseGuestFileAttributes {
 	return &f.GuestPosixFileAttributes
 }
+
+func (d *Driver) SetMachineOSFromFlags(flags drivers.DriverOptions) error {
+	d.OS = strings.ToLower(flags.String("--vmwarevsphere-os"))
+	if d.OS == "" {
+		d.OS = defaultMachineOS
+		return nil
+	}
+	err := d.CheckMachineOS(flags.String(d.OS))
```

Comment:
Is this correct? It seems to me it should be
```suggestion
	err := d.CheckMachineOS(d.OS)
```

Reply:
good catch - fixing this now.